screens/Authentication/SignUp/CreateCredentials.tsx

```diff
@@ -0,0 +1,189 @@
+import CredentialsButton from "@/components/CredentialsButton"
+import Badge from "@/components/ui/Badge"
+import CustomText from "@/components/ui/CustomText"
+import Input from "@/components/ui/Input"
+import CrossMark from "@/components/vectors/CrossMark"
+import { VALIDATION_PATTERNS } from "@/constants/patterns"
+import { RootStackParamList } from "@/navigation"
+import { theme } from "@/theme"
+import { AccountCreationStep } from "@/types"
+import { useNavigation } from "@react-navigation/native"
+import { NativeStackNavigationProp } from "@react-navigation/native-stack"
+import { useEffect, useMemo, useState } from "react"
+import {
+  StyleSheet,
+  View,
+  TouchableOpacity,
+  KeyboardAvoidingView,
+  Platform,
+} from "react-native"
+
+type CredentialsScrenProps = NativeStackNavigationProp<
+  RootStackParamList,
+  "EditSignal"
+>
+
+export default function CreateCredentials() {
+  const navigation = useNavigation<CredentialsScrenProps>()
+  const [step, setStep] = useState<AccountCreationStep>(1)
+  const [text, setText] = useState("")
+  const [isLoading, setIsLoading] = useState(false)
+  const [isError, setIsError] = useState(false)
+
+  const isInputValid = useMemo(() => {
+    if (text.trim().length < 5) return false
+
+    if (step === 1 && VALIDATION_PATTERNS.fullName.test(text)) return true
+    if (step === 2 && VALIDATION_PATTERNS.username.test(text)) return true
+
+    return false
+  }, [step, text])
+
+  const isDisabled = useMemo(() => {
+    if (isLoading || isError) return true
+    return !isInputValid
+  }, [isError, isInputValid, isLoading])
+
+  const stepsData: Record<
+    AccountCreationStep,
+    {
+      badgeName: string
+      titleText: string
+      inputPlaceholder: string
+      descriptionText: string
+    }
+  > = {
+    1: {
+      badgeName: "3/4",
+      titleText: "What is your name?",
+      inputPlaceholder: "Full name",
+      descriptionText: "Add your name so friends can find you.",
+    },
+    2: {
+      badgeName: "4/4",
```

Comment:
shouldn't the badge names be, `1/2` and `1/2` , respectively since we have only two steps?